src/Algebra/Module/Properties/Semimodule.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of semimodules.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                using (CommutativeSemiring)
+open import Algebra.Module.Bundles using (Semimodule)
+open import Level                  using (Level)
+
+module Algebra.Module.Properties.Semimodule
+  {r ℓr m ℓm : Level}
+  {semiring  : CommutativeSemiring r ℓr}
+  (semimod   : Semimodule semiring m ℓm)
+  where
+
+open CommutativeSemiring semiring
+open Semimodule          semimod
+open import Relation.Nullary.Negation using (contraposition)
+open import Relation.Binary.Reasoning.Setoid ≈ᴹ-setoid
+
+s≈0⇒s*v≈0 : ∀ {s} {v} → s ≈ 0# → s *ₗ v ≈ᴹ 0ᴹ
+s≈0⇒s*v≈0 {s} {v} s≈0 = begin
+  s  *ₗ v ≈⟨ *ₗ-congʳ s≈0 ⟩
+  0# *ₗ v ≈⟨ *ₗ-zeroˡ v ⟩
+  0ᴹ      ∎
+
+v≈0⇒s*v≈0 : ∀ {s} {v} → v ≈ᴹ 0ᴹ → s *ₗ v ≈ᴹ 0ᴹ
```

Comment:
Incredibly minor but we can save two characters in each type by merging the implicit brackets, e.g. `{s v}`.

Also is there a reason that you've switched from using the names `x` and `y` for elements of the main carrier set in the other module to `v` here?

Reply:
Changed as requested in next push.

The `s` and `v` are intended to suggest: _scalar_ and _vector_.

Reply:
This wasn't changed in the next push. I think consistency is the best, so could we change both the names of the variables in the type and the name of the lemma?

Reply:
Code changed, as requested:

- 's' => 'x'
- 'v' => 'y'

in next push.